core/base/inc/TSigHandling.h

```diff
@@ -0,0 +1,94 @@
+// @(#)root/base:$Id$
+// Author: Fons Rademakers   15/09/95
+
+/*************************************************************************
+ * Copyright (C) 1995-2000, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TSigHandling
+#define ROOT_TSigHandling
+
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// TSigHandling                                                         //
+//                                                                      //
+// Abstract base class defining a generic interface to the underlying   //
+// Operating System.                                                    //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef __CINT__
```

Comment:
In version 6 of ROOT (the master), we have replaced CINT with Cling which uses clang/llvm for the parsing and support full C++.  In version 6 __CINT__ is never defined :)